app/ai/voice/agents/breeze_buddy/utils/ivr.py

```
@@ -0,0 +1,335 @@
+        voice_name: TTS voice to use (default "sara")
+        ivr_greeting: Greeting prefix for IVR menu (default "Welcome")
+
+    Returns:
+        Audio bytes ready to send, or None if failed
+    """
+    try:
+        redis = await get_redis_service()
+
+        # Generate cache key from template names
+        names = "|".join(sorted([t["name"] for t in templates]))
+        cache_key = f"{IVR_AUDIO_CACHE_PREFIX}{hashlib.md5(names.encode()).hexdigest()}"
```

Ruff (0.14.14)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)